Patch the mozilla nsIUpdateService to process ActiveState notification messages
(bug 66793).

Index: /home/toddw/as/komodo-devel_clean/mozilla/build/moz700-ko7.8/mozilla/toolkit/mozapps/update/nsUpdateService.js
--- a/toolkit/mozapps/update/nsUpdateService.js	Thu Sep 29 11:24:41 2011 -0700
+++ b/toolkit/mozapps/update/nsUpdateService.js	Thu Jul 05 15:27:31 2012 -0700
@@ -2193,17 +2193,33 @@
     }
 
     const ELEMENT_NODE = Ci.nsIDOMNode.ELEMENT_NODE;
     var updates = [];
     for (var i = 0; i < updatesElement.childNodes.length; ++i) {
       var updateElement = updatesElement.childNodes.item(i);
-      if (updateElement.nodeType != ELEMENT_NODE ||
-          updateElement.localName != "update")
+      if (updateElement.nodeType != ELEMENT_NODE)
         continue;
-
       updateElement.QueryInterface(Ci.nsIDOMElement);
+
+      if (updateElement.localName == "notification") {
+        let notificationObj = {};
+        let attrs = ["action_label", "action_url", "id", "message", "title", "type"];
+        attrs.forEach(function(attr) {
+          notificationObj[attr] = (updateElement.hasAttribute(attr)
+                                   ? updateElement.getAttribute(attr)
+                                   : null);
+        });
+        Services.obs.notifyObservers(null,
+                                     "autoupdate-notification",
+                                     JSON.stringify(notificationObj));
+        continue;
+      }
+
+      if (updateElement.localName != "update")
+        continue;
+
       try {
         var update = new Update(updateElement);
       } catch (e) {
         LOG("Checker:updates get - invalid <update/>, ignoring...");
         continue;
       }
